lenses_python/SqlHandler.py

```diff
@@ -45,69 +47,17 @@ def _ValidateSqlQuery(self):
                  raise Exception('An error occurred while trying to validate sql query. Received response with '
                                         '\status code [{}] and text [{}]'.format(response.status_code, response.text))
 
-
-
-    def _with_requests(self, url):
-        """
-        Use GET request to get the data
-        :param url:
-        :return:
-        """
-        return get(url, params=self.params, headers=self.default_headers_2, stream=True)
-
-    def ExecuteSqlQuery(self, extract_pandas):
-        """
-        For SSE protocol use https://pypi.python.org/pypi/sseclient-py
-        ticket LEN-134
-        :param extract_pandas:
-        :return:if extract_pandas is empty return a list of dictionaries , otherwise return a pandas dataframe
-        """
-
-        EXECUTE_SQL_QUERY = "/api/sql/data"
-        url = self.url+EXECUTE_SQL_QUERY
-        response = self._with_requests(url)
-        client = sseclient.SSEClient(response)
-        messages = []
-        offset = []
-        data = {}
-        for event in client.events():
-            msg = event.data
-            if msg[0] == "0":
-                # Empty messages are just heartbeat
-                pass
-            else:
-                if msg[0] == "2":
-                    # 2 is offset details
-                    break
-                    # offset.append(msg[1:])
-                elif msg[0] == "3":
-                    # 3 is an error entry
-                    raise Exception("{}".format(msg[1:]))
-                elif msg[0] == "4":
-                    offset.append(msg[1:])
-                else:
-                    if msg[0] == "1":
-                        # 1 is a entry
-                        messages.append(loads(msg[1:]))
-        data["messages"] = messages
-        data["offset"] = offset
-        if extract_pandas == 0:
-            # In this case return a dictionary with two keys , one is the messages and the other one is the offset
-            return data
-        else:
-            # In this case we parse to ConvertToDF the messages and the we get the value of every message
-            # next we return a pandas data frame of them
-            return self._ConvertToDF(data["messages"])
-
-
     def _ConvertToDF(self, data):
         """
         Get data from sql handler and extract from generate dict the messages and then the dict-value from each one
 
         :param data: list of dictionaries
         :return: pandas dataframe
         """
-        data = list(map(lambda x: loads(x["value"]), data))
+        if type(data[0]["value"]) != type(dict()):
```

Comment:
this `data[0]` seems not safe. are you sure that always there will be at least one item?

Reply:
It must be at least one item , otherwise we can convert to pandas dataframe. So , I add check in browsing data, when the "data_list" 's length is 0 will raise exception of empty data list, something like that